planner/core/planbuilder.go

```diff
@@ -2842,6 +2842,8 @@ func splitWhere(where ast.ExprNode) []ast.ExprNode {
 		}
 	case *ast.ParenthesesExpr:
 		conditions = append(conditions, splitWhere(x.Expr)...)
+	case *ast.PatternLikeExpr:
```

Comment:
`splitWhere` is used for `select * from t where a = 1 and b=2`, it will split `a=1 and b=2` into a list that contains `[a=1, b=2]`, so I think no need change this